src/shared/components/dateRangePicker/NewDatePicker.tsx

```diff
@@ -0,0 +1,512 @@
+import React, {FC, useState, useContext, useEffect, useCallback} from 'react'
```

Comment:
nit: this component is pretty big and pretty dense. nothing specifically bad about how it is currently, just historically / stylistically larger components tend to mix dependencies within the business logic more than they may need to. if there's a way that breaking up the component also breaks up the `handle` logic, i'd float towards that.